src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala

```diff
@@ -169,7 +185,10 @@ object DockerPlugin extends AutoPlugin {
         Seq(makeWorkdir(dockerBaseDirectory)) ++
         (strategy match {
           case DockerPermissionStrategy.MultiStage =>
-            Seq(makeCopyFrom(dockerBaseDirectory, stage0name, user, group))
+            val layerIdsAscending = (dockerLayerMappings in Docker).value.map(_._1).distinct.sorted
+            layerIdsAscending.map { layerId =>
+              makeCopyFrom(s"/$layerId$dockerBaseDirectory", dockerBaseDirectory, stage0name, user, group)
+            }
           case DockerPermissionStrategy.Run =>
             Seq(makeCopy(dockerBaseDirectory), makeChmodRecursive(dockerChmodType.value, Seq(dockerBaseDirectory))) ++
               (addPerms map { case (tpe, v) => makeChmod(tpe, Seq(v)) })
```

Comment:
Depending on `dockerPermissionStrategy`, we do `chmod`s either in `stage0` (using layered paths) or in `stage1` (using paths without layers) - and `dockerAdditionalPermissions` in its current form can't support that. One possible solution would be to post-process `addPerms` in `stage0` to inject layer directories.

Reply:
I have code to solve this, just it wasn't 100% ready yesterday night to be pushed.